src/Tribe/I18n.php

```diff
@@ -0,0 +1,217 @@
+	 *
+	 * @todo Include support for the `load_theme_textdomain` + `load_muplugin_textdomain`
+	 */
+	public function get_i18n_strings_for_domains( $strings, $languages, $domains = array( 'default' ) ) {
+		sort( $languages );
+		$strings_buffer = [ $strings ];
+
+		foreach ( $languages as $language ) {
+			// Override the current locale w/ the one we need to compile the translations.
+			$language_strings = $this->with_locale(
+				$language,
+				[ $this, 'compile_translations' ],
+				[ $strings, $domains ]
+			);
+			$strings_buffer[] = $language_strings;
+		}
+
+		$strings = count( $strings_buffer ) > 1
+			? array_merge_recursive( ... $strings_buffer )
+			: reset( $strings_buffer );
+
+		// Prevent empty strings and duplicates.
+		foreach ( $strings as $key => $value ) {
+			$strings[ $key ] = array_filter(
+				array_unique(
+					array_map( 'sanitize_title_with_dashes', (array) $value )
+				)
+			);
+		}
+
+		return $strings;
+	}
+
+	/**
+	 * Executes a callback ensuring the `current_locale` will be set to the specified language code.
+	 *
+	 * The method will backup and detach the functions and methods currently filtering the `locale` filter to execute
+	 * the callback in isolation and restore the filters after that.
+	 * The main purpose of this method is to avoid a rat race against plugins and themes that will filter the locale
+	 * by attaching the filtering method or function at `PHP_INT_MAX`.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $locale The locale to set for the execution of the callback.
+	 * @param callable $do     The callable to execute in the context of a specific locale.
+	 * @param array    $args   A set of arguments that will be passed to the callback.
+	 *
+	 * @return mixed The callback return value, if any.
+	 */
+	protected function with_locale( $locale, callable $do, array $args = [] ) {
+		global $wp_filter;
+		$locale_filters_backup = isset( $wp_filter['locale'] ) ? $wp_filter['locale'] : [];
+		$wp_filter['locale']   = $locale_filters_backup instanceof \WP_Hook ? new \WP_Hook() : [];
+
+		$force_locale = static function () use ( $locale ) {
+			return $locale;
+		};
+
+		add_filter( 'locale', $force_locale );
+		$result = $do( ...$args );
+		remove_filter( 'locale', $force_locale );
+
+		// Restore the `locale` filtering functions.
+		$wp_filter['locale'] = $locale_filters_backup;
+
+		return $result;
+	}
+
+	/**
+	 * Compiles the translations for a set of strings iterating on a set of domains.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,array|string> $strings The set of strings to compile the translations for.
+	 * @param string|array<string>       $domains The domain(s) that should be used to compile the string translations.
+	 *
+	 * @return array<string|array> A map of the compiled string translations.
+	 */
+	 public function compile_translations( array $strings, $domains ) {
+		$cache_salts = [ $strings, $domains, get_locale() ];
```

Comment:
* :warning: Found precision alignment of 1 spaces.
* :no_entry_sign: Line indented incorrectly; expected 1 tabs, found 1

<sub>hash: b9375d4ac762cce528831dc40ac4ed40</sub>